client/app/src/api/stub.js

```diff
@@ -0,0 +1,151 @@
+import { sendChoice } from '../state/actions';
+import { swatchColors } from '../constants/Colors';
+import { CHOICE_COLOR, CHOICE_CHILLS } from '../constants/Choices';
+
+const colors = swatchColors.flat();
+const now = new Date();
+
+const generateStageEnd = () => {
+  const eventData = {
+    stageId: 'STAGE_END',
+    songs: [
+      {
+        displayName: 'Color Song 1',
+        startTime: new Date(new Date().setMinutes(now.getMinutes() - 34)).toISOString(),
```

Comment:
JS ignorance question: what's the reason for the wrapped `Date` objects for all of these?

Reply:
tl;dr: non-fluid JS syntax

The end result I want is an ISO string, which you get by calling `.toISOString()` on a Date object. So you should exect the final form to look something like:
```js
date.toISOString()
```

So, how to get the necessary `date` object?  Well, I want to add and remove some time from a fixed time point (`now`). Can't just do `now + 180`, that returns a meaningless string. I can use `.setMinutes()`, but that modifies `now`. I don't want to modify `now`, because I'm reusing that elsewhere. So I need a new object. That gives us something like
```js
new Date().setMinutes(now.getMinutes() - 3).toISOString()
```

_Except_ `.setMinutes()` returns a number (the unix epoch), not a date object. So I can't chain a call to `.toISOString()` without first wrapping the result in another date object. That gives us

```js
new Date(new Date().setMInutes(now.getMintues() - 3)).toISOString()
```

An alternative would be
```js
const before = new Date();
before.setMinutes(now.getMinutes() - 3);
return before.toISOString();
```

But that's multiple lines. I could do
```js
new Date(now.valueOf() - 180000).toISOString()
```

But I think that's less semantic. In short, I haven't found a better way of doing it using the syntax available in vanilla JS. It's something that's bothered me for a while.